public/app/features/plugins/loader/cache.test.ts

```diff
@@ -1,54 +1,64 @@
-    const address = `http://localhost:3000/public/${path}.js`;
-
-    registerPluginInCache({ path, version });
-
-    const url = `${address}?_cache=${encodeURI(version)}`;
-    expect(resolveWithCache(address, now)).toBe(url);
+    it('should not register a plugin if it already exists in the cache', () => {
+      const pluginId = 'plugin2';
+      const plugin = { pluginId, version: '2.0.0' };
+      registerPluginInCache(plugin);
+      const plugin2 = { pluginId, version: '2.5.0' };
+      registerPluginInCache(plugin2);
+      expect(getPluginFromCache(pluginId)?.version).toBe('2.0.0');
+    });
   });
 
-  it('should append Date.now as cache flag if plugin is not registered in buster', () => {
-    const slug = 'bubble-chart-2';
-    const address = `http://localhost:3000/public/plugins/${slug}/module.js`;
+  describe('invalidatePluginInCache', () => {
+    it('should invalidate a plugin in the cache', () => {
+      const pluginId = 'plugin3';
+      const plugin = { pluginId, version: '3.0.0' };
+      registerPluginInCache(plugin);
+      invalidatePluginInCache(pluginId);
+      expect(getPluginFromCache(pluginId)).toBeUndefined();
+    });
 
-    const url = `${address}?_cache=${encodeURI(String(now))}`;
-    expect(resolveWithCache(address, now)).toBe(url);
+    it('should not throw an error if the plugin does not exist in the cache', () => {
+      expect(() => invalidatePluginInCache('nonExistentPlugin')).not.toThrow();
+    });
   });
 
-  it('should append Date.now as cache flag if plugin is invalidated in buster', () => {
-    const slug = 'bubble-chart-3';
-    const version = 'v1.0.0';
-    const path = `/public/plugins/${slug}/module.js`;
-    const address = `http://localhost:3000/public/${path}.js`;
-
-    registerPluginInCache({ path, version });
-    invalidatePluginInCache(slug);
+  describe('resolveWithCache', () => {
+    it('should resolve URL with timestamp cache bust parameter', () => {
+      const url = 'http://localhost:3000/public/plugins/plugin4/module.js';
+      expect(resolveWithCache(url)).toContain('_cache=123456');
+    });
 
-    const url = `${address}?_cache=${encodeURI(String(now))}`;
-    expect(resolveWithCache(address, now)).toBe(url);
+    it('should resolve URL with plugin version as cache bust parameter if available', () => {
+      const plugin = { pluginId: 'plugin5', version: '5.0.0' };
+      registerPluginInCache(plugin);
+      const url = 'http://localhost:3000/public/plugins/plugin5/module.js';
+      expect(resolveWithCache(url)).toContain('_cache=5.0.0');
+    });
   });
 
-  it('should also clear plugin settings cache', () => {
-    const slug = 'bubble-chart-3';
-    const version = 'v1.0.0';
-    const path = `/public/plugins/${slug}/module.js`;
-
-    const clearPluginSettingsCacheSpy = jest.spyOn(pluginSettings, 'clearPluginSettingsCache');
-
-    registerPluginInCache({ path, version });
-    invalidatePluginInCache(slug);
+  describe('getPluginFromCache', () => {
```

Comment:
I would probably also add a test-case for `encodeURI(version)` part as well (to highlight in which scenarios it is necessary).